src/Game.js

```diff
@@ -0,0 +1,60 @@
+class Game {
+
+  constructor() {
+    this._rolls = [];
+    this._gameover = false;
+  }
+
+  roll(pins) {
+    this._rolls.push(pins);
+  }
+
+  score() {
+    let result = 0;
+    let rollindex = 0;
+    let game = this;
+
+    for (let frame = 0; frame < 10; frame ++) {
+      if (strike()) {
+        result += calculateStrikeScore();
+        rollindex ++;
+      } else if (spare()) {
+        result += calculateSpareScore();
+        rollindex += 2;
+      } else {
+        result += calculateScore();
+        rollindex += 2;
+      }
+    }
+
+    function endGame() {
```

Comment:
I'm loving the shorthand use of methods & classes, simplifies the code and makes it less smelly ;) - the use of let within the if else statement cleans up the code very nicely as well - saves having to pass frame for every method!